python/cris_import.py

```diff
@@ -0,0 +1,257 @@
+import csv
+from datetime import datetime
+import os
+import time
+from zoneinfo import ZoneInfo
+
+import requests
+
+from columns import COLUMNS
+
+FILE_DIR = "cris_csvs"
+HASURA_ENDPOINT = "http://localhost:8084/v1/graphql"
+UPLOAD_BATCH_SIZE = 1000
+
+CRASH_UPSERT_MUTATION = """
+mutation UpsertCrashes($objects: [db_crashes_cris_insert_input!]!) {
+  insert_db_crashes_cris(
+    objects: $objects, 
+    on_conflict: {
+        constraint: crashes_cris_pkey,
+        update_columns: [$updateColumns]
+    }) {
+    affected_rows
+  }
+}
+"""
+
+UNIT_UPSERT_MUTATION = """
+mutation UpsertUnits($objects: [db_units_cris_insert_input!]!) {
+  insert_db_units_cris(
+    objects: $objects, 
+    on_conflict: {
+        constraint: unique_units_cris,
+        update_columns: [$updateColumns]
+    }) {
+    affected_rows
+  }
+}
+"""
+
+PERSON_UPSERT_MUTATION = """
+mutation UpsertPeople($objects: [db_people_cris_insert_input!]!) {
+  insert_db_people_cris(
+    objects: $objects, 
+    on_conflict: {
+        constraint: unique_people_cris,
+        update_columns: [$updateColumns]
+    }) {
+    affected_rows
+  }
+}"""
+
+mutations = {
+    "crashes": CRASH_UPSERT_MUTATION,
+    "units": UNIT_UPSERT_MUTATION,
+    "persons": PERSON_UPSERT_MUTATION,
+}
+
+
+def make_upsert_mutation(table_name):
+    """Sets the column names in the upsert on conflict array"""
+    upsert_mutation = mutations[table_name]
+    updateColumns = [
+        col["name"]
+        for col in COLUMNS
+        if col["table_name"] == table_name
+        and col["source"] == "cris"
```

Comment:
this is rad, and it gives us the power control and document exactly which columns can be overloaded right at the beginning of the process instead of relying on the client application code or other places further down the line that can become inconsistent over time.